### PR TITLE
[Core] Extend ContainerExpression to char and int

### PR DESCRIPTION
**Pull request by sunethwarna**
**📝 Description**
This PR extends `LiteralFlatExpression` to `char` and `int` types, and `LiteralExpression` to int types. These extensions are accordingly followed in `ContainerExpression` and `SpecializedContainerExpression`.

-- TO DO:

- [x] Add Tests

**🆕 Changelog**
- Extends `LiteralFlatExpression` to `char` and `int` types
- Extends `LiteralExpression` to `int` type.
- Adapt `ContainerExpression` and `SpecializedContainerExpression` to support new types

### CODE REVIEW

File: kratos/containers/container_expression/expressions/literal/literal_flat_expression.h [hunk 1]
```diff
@@ -119,33 +131,39 @@ class KRATOS_API(KRATOS_CORE) LiteralFlatExpression : public Expression {
          *
```
Comment (matekelemen) [inline kratos/containers/container_expression/expressions/literal/literal_flat_expression.h:131]: Looks like microsoft, in their infinite wisdom, disabled review comments outside the vicinity of changes, so I'll just put it here:
In lines 114 and 115 (new line indices) update `double` to `data type` or something like that to reflect that `Expression` is now a template.

File: kratos/containers/container_expression/expressions/literal/literal_flat_expression.cpp [hunk 2]
```diff
@@ -20,80 +20,117 @@
...
-void LiteralFlatExpression::SetData(
+template<class TRawDataType>
+void LiteralFlatExpression<TRawDataType>::SetData(
     const IndexType EntityDataBeginIndex,
     const IndexType ComponentIndex,
-    const double Value)
+    const TRawDataType Value)
 {
-    mData[EntityDataBeginIndex + ComponentIndex] = Value;
+    *(mData.DataBegin() + EntityDataBeginIndex + ComponentIndex) = Value;
```
Comment (matekelemen) [inline kratos/containers/container_expression/expressions/literal/literal_flat_expression.cpp]: what's wrong with square brackets?

File: kratos/containers/container_expression/container_expression.cpp [hunk 3]
```diff
@@ -106,10 +106,76 @@ const ModelPart::ConditionsContainerType& GetContainer<ModelPart::ConditionsCont
...
+
+    auto p_expression = LiteralFlatExpression<TRawDataType>::Create(number_of_entities, shape);
+    rContainerExpression.SetExpression(p_expression);
+    TRawDataType* data_itr = p_expression->DataBegin();
+
+    const IndexType flattened_size = rContainerExpression.GetExpression().GetFlattenedShapeSize();
+    const IndexType total_size = number_of_entities * flattened_size;
+
+    IndexPartition<IndexType>(total_size).for_each([pBegin, data_itr](const IndexType Index) {
+        *(data_itr + Index) = *(pBegin + Index);
```
Comment (matekelemen) [inline kratos/containers/container_expression/container_expression.cpp]: square brackets?

File: kratos/containers/container_expression/specialized_container_expression_impl.h [hunk 4]
```diff
@@ -54,11 +54,13 @@ void SpecializedContainerExpression<TContainerType, TContainerDataIO, TMeshType>
     const auto& r_container = this->GetContainer();
     const IndexType number_of_entities = r_container.size();
 
+    using raw_data_type = std::conditional_t<std::is_same_v<TDataType, int>, int, double>;
```
Comment (matekelemen) [inline kratos/containers/container_expression/specialized_container_expression_impl.h]: um, what? Why? What about chars?

File: kratos/containers/container_expression/expressions/literal/literal_flat_expression.cpp (hunk 2 above)
Comment (sunethwarna) [reply kratos/containers/container_expression/expressions/literal/literal_flat_expression.cpp]: good idea.... too much pointer usage i guess ;)

File: kratos/containers/container_expression/specialized_container_expression_impl.h (hunk 4 above)
Comment (sunethwarna) [reply kratos/containers/container_expression/specialized_container_expression_impl.h]: We don't have variables of type `char`, thats why I was omitted the `char` in here. But, now did it anyways :)

File: kratos/containers/container_expression/expressions/literal/literal_flat_expression.h (hunk 1 above)
Comment (sunethwarna) [reply kratos/containers/container_expression/expressions/literal/literal_flat_expression.h:131]: done :)

File: kratos/containers/container_expression/expressions/literal/literal_flat_expression.cpp (hunk 2 above)
Comment (sunethwarna) [reply kratos/containers/container_expression/expressions/literal/literal_flat_expression.cpp]: done :)

File: kratos/containers/container_expression/container_expression.cpp (hunk 3 above)
Comment (sunethwarna) [reply kratos/containers/container_expression/container_expression.cpp]: done :)

File: kratos/containers/container_expression/expressions/literal/literal_flat_expression.cpp (hunk 2 above)
Comment (sunethwarna) [reply kratos/containers/container_expression/expressions/literal/literal_flat_expression.cpp]: Actually it does not work since mData is a struct.

**Comment by sunethwarna** [comment]
@matekelemen could you check and re-approve? I applied your suggestions :)

**Review by matekelemen — APPROVED** [review]

**Pull request merged**